### PR TITLE
Basic example of reversing the digits in a integer.

### PR DESCRIPTION
**Pull request by harrisonbrock**

### CODE REVIEW

File: topics/go/algorithms/numbers/reverse/reverse.go [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package numbers
+
+// Reverse takes in a integer and returns a the integer in reverse order.
```
Comment (ardan-bkennedy) [inline topics/go/algorithms/numbers/reverse/reverse.go:3]: Read the comment.

File: topics/go/algorithms/numbers/reverse/reverse.go [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+func Reverse(num int) int {
+
+	// Set result to zero.
```
Comment (ardan-bkennedy) [inline topics/go/algorithms/numbers/reverse/reverse.go:6]: Construct result to its zero value.

File: topics/go/algorithms/numbers/reverse/reverse.go [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+	var result int
+
+	// Loop until num is not zero.
```
Comment (ardan-bkennedy) [inline topics/go/algorithms/numbers/reverse/reverse.go:9]: Why? When would num not be zero?

File: topics/go/algorithms/numbers/reverse/reverse.go [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+	for num != 0 {
+
+		// Get the last digit from num.
```
Comment (ardan-bkennedy) [inline topics/go/algorithms/numbers/reverse/reverse.go:12]: // Perform a modulus operation to get the last digit from the value set in num. 
// https://www.geeksforgeeks.org/find-first-last-digits-number/

File: topics/go/algorithms/numbers/reverse/reverse.go [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+		// Example: 125 % 10 = 5.
+		last := num % 10
+
+		// Move result one place to the left and add last.
```
Comment (ardan-bkennedy) [inline topics/go/algorithms/numbers/reverse/reverse.go:16]: // Multiple the current result by 10 to shit the digits in
// the current result to the left.
result = result*10

// Add the digit we took from the end of num to the result.
result += last

Break these two lines into different steps and explain it

File: topics/go/algorithms/numbers/reverse/reverse.go [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+		result = result*10 + last
+
+		// Remove the right most digit from num.
```
Comment (ardan-bkennedy) [inline topics/go/algorithms/numbers/reverse/reverse.go:19]: // Remove the digit we just reversed from num.

File: topics/go/algorithms/numbers/reverse/reverse.go [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+		// Example: 125 / 10 = 12.
+		num = num / 10
+	}
+	return result
```
Comment (ardan-bkennedy) [inline topics/go/algorithms/numbers/reverse/reverse.go:23]: I think I want to find a way to show the example so people can see how the algorithm works. Maybe we show each interation 
https://play.golang.org/p/Gy33iB6wGs0

last[5] = num[125] % 10
last[2] = num[12] % 10
last[1] = num[1] % 10

result[0] = result[0] * 10
result[50] = result[5] * 10
result[520] = result[52] * 10

result[5] = result[0] += last[5]
result[52] = result[50] += last[2]
result[521] = result[520] += last[1]

num[12] = num[125] / 10
num[1] = num[12] / 10
num[0] = num[1] / 10

Comment (harrisonbrock) [reply topics/go/algorithms/numbers/reverse/reverse.go:23]: Should we create a main.go to show this? Or include it in reverse.go?

Comment (ardan-bkennedy) [reply topics/go/algorithms/numbers/reverse/reverse.go:26]: Somehow in this file or we add a doc.go

**Pull request merged**